Content.Server/Nyanotrasen/Roboisseur/RoboisseurSystem.cs

```diff
@@ -0,0 +1,355 @@
+            foreach (var roboisseur in EntityQuery<RoboisseurComponent>())
+            {
+                roboisseur.Accumulator += frameTime;
+                roboisseur.BarkAccumulator += frameTime;
+                if (roboisseur.BarkAccumulator >= roboisseur.BarkTime.TotalSeconds)
+                {
+                    roboisseur.BarkAccumulator = 0;
+                    string message = Loc.GetString(_random.Pick(DemandMessages), ("item", roboisseur.DesiredPrototype.Name));
+                    if (roboisseur.ResetTime.TotalSeconds - roboisseur.Accumulator < 120)
+                    {
+                        roboisseur.Impatient = true;
+                        message = Loc.GetString(_random.Pick(ImpatientMessages), ("item", roboisseur.DesiredPrototype.Name));
+                    }
+                    else if (CheckTier(roboisseur.DesiredPrototype.ID) > 2)
+                        message = Loc.GetString(_random.Pick(DemandMessagesTier2), ("item", roboisseur.DesiredPrototype.Name));
+                    _chat.TrySendInGameICMessage(roboisseur.Owner, message, InGameICChatType.Speak, false);
+                }
+
+                if (roboisseur.Accumulator >= roboisseur.ResetTime.TotalSeconds)
+                {
+                    roboisseur.Impatient = false;
+                    NextItem(roboisseur);
+                }
+            }
+        }
+
+        private void RewardServicer(EntityUid uid, RoboisseurComponent component, int tier)
+        {
+            Random r = new Random();
+            int rewardToDispense = r.Next(100, 350) + 250 * tier;
+
+            _material.SpawnMultipleFromMaterial(rewardToDispense, "Credit", Transform(uid).Coordinates);
+            if(tier > 1){
+                while (tier != 0)
+                {
+                    EntityManager.SpawnEntity(_random.Pick(RobossuierRewards), Transform(uid).Coordinates);
+                    tier--;
+                }
+            }
+            return;
+        }
+
+        private void OnInteractHand(EntityUid uid, RoboisseurComponent component, InteractHandEvent args)
+        {
+            if (!TryComp<ActorComponent>(args.User, out var actor))
+                return;
+            string message = Loc.GetString(_random.Pick(DemandMessages), ("item", component.DesiredPrototype.Name));
+            if (CheckTier(component.DesiredPrototype.ID) > 1)
+                message = Loc.GetString(_random.Pick(DemandMessagesTier2), ("item", component.DesiredPrototype.Name));
+            _chat.TrySendInGameICMessage(component.Owner, message, InGameICChatType.Speak, false);
+        }
+
+        private void OnInteractUsing(EntityUid uid, RoboisseurComponent component, InteractUsingEvent args)
+        {
+            if (HasComp<MobStateComponent>(args.Used))
+                return;
+
+            if (!TryComp<MetaDataComponent>(args.Used, out var meta))
+                return;
+
+            if (meta.EntityPrototype == null)
+                return;
```

Comment:
I don't see a reason for MobStateComponent to be checked here, remove it if there isn't a reason.

`MetaDataComponent` and `TransformComponent` are always on every entity, there is no need for a TryGet or HasComp. Getting these two components can also be done via functions in the system, `MetaData(EntityUid)` and `Transform(EntityUid)` respectively.
```suggestion
            if (HasComp<MobStateComponent>(args.Used) ||
                MetaData(args.Used)?.EntityPrototype == null)
                return;
```